clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -559,8 +606,10 @@ private void expectJoinGroup(
                 return false;
             }
             JoinGroupRequestData joinGroupRequest = ((JoinGroupRequest) body).data();
+            boolean isReasonMatching = expectedReason == null || joinGroupRequest.reason().equals(expectedReason);
```

Comment:
I don't think the logic here is correct. 
If the `expectedReason` is null, but the `joinGroupRequest.reason()` is not null, should it be passed?
That also implies you need add tests to cover this situation.

Reply:
i added a comment. the reason is never set to null in AbstractCoordinator.